src/careamics/models/lvae/noise_models.py

```diff
@@ -168,19 +210,20 @@ class GaussianMixtureNoiseModel(nn.Module):
             While initializing a new object of the class `GaussianMixtureNoiseModel` from scratch, set this to `None`.
     """
 
-    def __init__(self, **kwargs):
+    # TODO training a NM relies on getting a clean data(N2V e.g,)
+    def __init__(self, config: GMNMModel):
         super().__init__()
         self._learnable = False
 
-        if kwargs.get("params") is None:
-            weight = kwargs.get("weight")
-            n_gaussian = kwargs.get("n_gaussian")
-            n_coeff = kwargs.get("n_coeff")
-            min_signal = kwargs.get("min_signal")
-            max_signal = kwargs.get("max_signal")
+        if config.path is None:
```

Comment:
Not a fan of this pattern. Maybe there should be an option to initialise the class with a path or a config object. If initialised with a config then just store as an instance attribute. If initialised with a path then a load the saved parameters and create the config object from these parameters to store as an attribute. (And path is removed as an attribute from the config). This would also solve my previous comment since the class can be initialised directly from the path.